lib/srv/desktop/windows_server.go

```diff
@@ -753,6 +755,20 @@ func (s *WindowsService) handleConnection(proxyConn *tls.Conn) {
 	}
 	desktop := desktops[0]
 
+	if desktop.NonAD() && buildType == modules.BuildOSS {
```

Comment:
This code runs as part of windows_desktop_service, but `CheckOSSDesktopsLimit` only works when run locally on auth, right? How is this going to work of w_d_s runs in a separate process?

Reply:
On separate service this goes through cache, right? My tests suggest that it should work just fine. I don't understand the cache layer very well, though, so it might be that I messed something up

Reply:
You don't need to do this check at all here. Let the AP enforce this by checking the cert (which is always generated by the auth server)